src/Vault.sol

```diff
@@ -118,14 +118,35 @@ contract Vault is IVault, VaultToken, Ownable {
     }
 
     /// @inheritdoc IVault
-    function settle(Currency currency) external payable override isLocked returns (uint256 paid) {
+    function settle(Currency currency) public payable override isLocked returns (uint256 paid) {
         uint256 reservesBefore = reservesOfVault[currency];
         reservesOfVault[currency] = currency.balanceOfSelf();
         paid = reservesOfVault[currency] - reservesBefore;
         // subtraction must be safe
         SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
     }
 
+    function settleAndRefund(Currency currency, address to)
+        external
+        payable
+        isLocked
+        returns (uint256 paid, uint256 refund)
+    {
+        paid = settle(currency);
```

Comment:
Should we calculate the amount we need to refund in advance? then just need to execute SettlementGuard.accountDelta once if have excess token.

```
function settleAndRefund(Currency currency, address to)
        external
        payable
        isLocked
        returns (uint256 paid, uint256 refund)
    {
        paid = currency.balanceOfSelf() - reservesOfVault[currency];
        // need to consider the case when the currencyDelta is negative ?.
        uint256 currentCurrencyDelta = (SettlementGuard.getCurrencyDelta(msg.sender, currency)).toUint256();
       
        if (paid > currentCurrencyDelta) {
            refund = paid - currentCurrencyDelta;
            paid = currentCurrencyDelta;
        }
        reservesOfVault[currency] += paid;
        SettlementGuard.accountDelta(msg.sender, currency, -(paid.toInt128()));
        if (refund > 0) {
            currency.transfer(to, refund);
        }
    }
```

Reply:
thanks! adopted this idea since it'll save 1 `sload` for refund case. though did a small tweak to handle negative currency delta as `settle()` wont revert with `SafeCastOverflow` for negative balance delta